packages/prettier/index.ts

```diff
@@ -52,82 +56,85 @@ export function create(
 	return {
 		name: 'prettier',
 		create(context): ServicePluginInstance {
+			const languages = options.languages ?? ['html', 'css', 'scss', 'typescript', 'javascript'];
 
 			let prettier: typeof import('prettier');
 			try {
 				prettier = options.prettier
 					?? options.getPrettier?.(context.env)
 					?? require('prettier');
-			} catch (e) {
-				throw new Error("Could not load Prettier: " + e);
-			}
-			const languages = options.languages ?? ['html', 'css', 'scss', 'typescript', 'javascript'];
 
```

Comment:
What about doing this:

```ts
let prettier: typeof import('prettier');
try {
	prettier = options.prettier
		?? options.getPrettier?.(context.env)
		?? require('prettier');
} catch (e) {
	if (!options.allowImportError) {
    throw new Error("Could not load Prettier: " + e);
  }
}

if (!prettier) {
  return {}
}
```

This way you don’t need to wrap so much logic in the the `try` block. Also you don’t need to check if `prettier` exists in every LSP call.

Or possibly even better:

```ts
let prettier = options.prettier
if (!prettier) {
  if (options.getPrettier) {
    prettier = options.getPrettier(context.env)
  } else {
    prettier = require('prettier')
  }
}

if (!prettier) {
  return {}
}
```

This allows `getPrettier` to return `undefined` as an alternative to adding the `allowImportError` option. I have my doubts if the fallback to `require('prettier')` is even desirable at all. I think no Prettier formatting is better than formatting with a wrong Prettier version.

Reply:
Both are perfectly fine with me! I went with the second one at first actually, but wasn't sure if we were still okay with doing breaking changes. I'd be happy with that solution too

Reply:
> Or possibly even better:
> 
> ```ts
> let prettier = options.prettier
> if (!prettier) {
>   if (options.getPrettier) {
>     prettier = options.getPrettier(context.env)
>   } else {
>     prettier = require('prettier')
>   }
> }
> 
> if (!prettier) {
>   return {}
> }
> ```

Let's do this. 👍